terraform/eval_state.go

```diff
@@ -43,10 +43,9 @@ func (n *EvalReadState) Eval(ctx EvalContext) (interface{}, error) {
 		if idx < 0 {
 			idx = len(rs.Tainted) - 1
 		}
-
-		if idx < len(rs.Tainted) {
+		if idx >= 0 && idx < len(rs.Tainted) {
 			// Return the proper tainted resource
-			result = rs.Tainted[n.TaintedIndex]
+			result = rs.Tainted[idx]
```

Comment:
Err, tests were failing suddenly without this change (core tests). I don't know, I really don't know.